File: 김금환/완전탐색/최소직사각형.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int maxOfMin = 0;
    int maxOfAll = 0;
    
    for (vector<int> size : sizes) {
        // 한 명함의 w, h 중 작은 값을 구하고, 이 값들 중에서 최댓값 구하기
        maxOfMin = max(maxOfMin, min(size[0], size[1]));
        // 모든 명함의 w, h 중에서 최댓값 구하기
        maxOfAll = max(maxOfAll, max(size[0], size[1]));
    }
    
    return maxOfMin * maxOfAll;
}

File: 안윤철/완전탐색/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;

        int length = sizes.length;

        int latMax = sizes[0][0];
        int lonMax = sizes[0][1];
        int area = latMax * lonMax;
        for (int i = 1; i < length; i++) {
            int lat = sizes[i][0];
            int lon = sizes[i][1];

            if (Math.max(latMax, lat) * Math.max(lonMax, lon) > Math.max(lonMax, lat) * Math.max(latMax, lon)) {
                area = Math.max(lonMax, lat) * Math.max(latMax, lon);
                lonMax = Math.max(lonMax, lat);
                latMax = Math.max(latMax, lon);
            } else {
                area = Math.max(latMax, lat) * Math.max(lonMax, lon);
                lonMax = Math.max(lonMax, lon);
                latMax = Math.max(latMax, lat);
            }
        }

        return area;
    }
}